### PR TITLE
1829 SR fixed, Tilelay

### PR DESCRIPTION
**Pull request by volker18xxdev**
in 1829 Tile Upgrade can done anywhere, Only 4 privates in the SR (draft.rb), and preparing buy-sell shares to include the steam pakets companies to be bought in later Stock rounds.

### CODE REVIEW

File: lib/engine/game/g_1829/step/buy_sell_par_shares.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+            player.cash >= company.value
+          end
+
+          def buyable_bank_owned_companies
+            @companies.select { |c| c.owner == @bank && c.revenue > 20 }
+          end
+
+          def available
+            @companies.select { |c| !c.closed? && c.owner == @bank && c.revenue > 20 }
```
Comment (tobymao) [inline lib/engine/game/g_1829/step/buy_sell_par_shares.rb:42]: why don't you call buyable banked ownedcompanies in all these places/ you have too many repeated lines of code

Comment (volker18xxdev) [reply lib/engine/game/g_1829/step/buy_sell_par_shares.rb]: done

Comment (tobymao) [reply lib/engine/game/g_1829/step/buy_sell_par_shares.rb]: there are many other references, please clean all of them up and fix the build

**Comment by volker18xxdev** [comment]
There are 2 type of companies, the first 4 (revenue <= 20) are sold in the SR, the rest (5) are Steam Pakets, which can be bought in the following Stockrounds as an alternative to the regular Shares. because of this i've 2 definitions.

**Comment by tobymao** [comment]
add closed? to the buyable_bank_owned_companies and replace all references

**Comment by volker18xxdev** [comment]
closed? added and 1 reference in draft.rb replaced.

**Comment by tobymao** [comment]
you didn't make the requested changes, i'd like you to remove the duplicate code

**Comment by volker18xxdev** [comment]
duplicate code is removed. sorry

File: lib/engine/game/g_1829/step/buy_sell_par_shares_companies.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+            actions << 'pass' unless actions.empty?
+            actions
+          end
+
+          def can_buy_any_companies?(entity)
+            return false if
+              @game.num_certs(entity) >= @game.cert_limit
+
+            @game.companies.any? { |c| c.owner == @bank && !c.closed? }
```
Comment (tobymao) [inline lib/engine/game/g_1829/step/buy_sell_par_shares_companies.rb:27]: duplicate

Comment (volker18xxdev) [reply lib/engine/game/g_1829/step/buy_sell_par_shares_companies.rb:27]: sorry,  I'm not so good in ruby. the first definition is in step/draft.rb. How can i access this definition from buy_sell_par_shares_companies.rb? Can you help me?

Comment (tobymao) [reply lib/engine/game/g_1829/step/buy_sell_par_shares_companies.rb:27]: you need to put the definition in game and then call it

**Comment by tobymao** [comment]
closing due to inactivity

**Pull request closed without merging**